package proto_ff;

import "yd_fieldoptions.proto";


message E_BattlepassBattlepass
{
	optional int32 m_id = 1[(yd_fieldoptions.field_cname) = "id"];
	optional int32 m_reward = 2[(yd_fieldoptions.field_cname) = "奖励组"];
	optional int32 m_task = 3[(yd_fieldoptions.field_cname) = "任务组"];
	optional string m_taskNum = 4[(yd_fieldoptions.field_cname) = "抽取任务数量", (yd_fieldoptions.field_bufsize) = 64];
	optional int32 m_freshCost = 5[(yd_fieldoptions.field_cname) = "刷新消耗"];
	optional int32 m_shop = 6[(yd_fieldoptions.field_cname) = "店铺组"];
	optional int32 m_shopCurrency = 7[(yd_fieldoptions.field_cname) = "商店货币"];
	optional int32 m_severOpenType = 8[(yd_fieldoptions.field_cname) = "服务器开放类型"];
	optional int32 m_openSeverDay = 9[(yd_fieldoptions.field_cname) = "开始开服时间"];
	optional int32 m_closeSeverDay = 10[(yd_fieldoptions.field_cname) = "关闭时间"];
	optional int32 m_priceType = 11[(yd_fieldoptions.field_cname) = "价格类型"];
	optional int32 m_price = 12[(yd_fieldoptions.field_cname) = "普通版价格"];
	optional int32 m_priceP = 13[(yd_fieldoptions.field_cname) = "高级版额外价格"];
	optional int32 m_presentExp = 14[(yd_fieldoptions.field_cname) = "高级版赠送的经验值"];
	optional int32 m_expCost = 15[(yd_fieldoptions.field_cname) = "经验货币"];
	optional int32 m_expPrice = 16[(yd_fieldoptions.field_cname) = "每货币兑换经验值"];
	optional int32 m_expItem = 17[(yd_fieldoptions.field_cname) = "获得道具自动获得经验"];
}


message Sheet_BattlepassBattlepass
{
	repeated E_BattlepassBattlepass E_BattlepassBattlepass_List = 1[(yd_fieldoptions.field_arysize)=16];
}

message E_BattlepassRewardItemDesc
{
	optional int32 m_num = 1[(yd_fieldoptions.field_cname) = "数量"];
	optional int32 m_id = 2[(yd_fieldoptions.field_cname) = "ID"];
}

message E_BattlepassRewardProitemDesc
{
	optional int32 m_num = 1[(yd_fieldoptions.field_cname) = "数量"];
	optional int32 m_id = 2[(yd_fieldoptions.field_cname) = "ID"];
}

message E_BattlepassReward
{
	optional int32 m_id = 1[(yd_fieldoptions.field_cname) = "id"];
	optional int32 m_groupid = 2[(yd_fieldoptions.field_cname) = "组类型"];
	optional int32 m_lv = 3[(yd_fieldoptions.field_cname) = "等级"];
	optional int32 m_exp = 4[(yd_fieldoptions.field_cname) = "升级所需经验"];
	optional int32 m_Core = 5[(yd_fieldoptions.field_cname) = "核心节点标识"];
	optional int32 m_buyLevel = 6[(yd_fieldoptions.field_cname) = "购买节点标识"];
	optional int32 m_showItem = 7[(yd_fieldoptions.field_cname) = "展示物id"];
	repeated E_BattlepassRewardItemDesc m_item = 8[(yd_fieldoptions.field_cname) = "奖励物品", (yd_fieldoptions.field_arysize) = 1];
	repeated E_BattlepassRewardProitemDesc m_proItem = 9[(yd_fieldoptions.field_cname) = "加强奖励物品", (yd_fieldoptions.field_arysize) = 2];
}


message Sheet_BattlepassReward
{
	repeated E_BattlepassReward E_BattlepassReward_List = 1[(yd_fieldoptions.field_arysize)=2048];
}

message E_BattlepassTask
{
	optional int32 m_id = 1[(yd_fieldoptions.field_cname) = "id"];
	optional int32 m_group = 2[(yd_fieldoptions.field_cname) = "组id"];
	optional int32 m_taskRefresh = 3[(yd_fieldoptions.field_cname) = "任务类型"];
	optional int32 m_lv = 4[(yd_fieldoptions.field_cname) = "开放等级"];
	optional int32 m_tag = 5[(yd_fieldoptions.field_cname) = "所属页签"];
	optional int32 m_eventType = 6[(yd_fieldoptions.field_cname) = "事件类型"];
	optional int32 m_reachPrice = 7[(yd_fieldoptions.field_cname) = "达成值"];
	optional int32 m_link = 8[(yd_fieldoptions.field_cname) = "链接"];
	optional int32 m_exp = 9[(yd_fieldoptions.field_cname) = "经验"];
	repeated int32 m_param = 10[(yd_fieldoptions.field_cname) = "事件完成参数", (yd_fieldoptions.field_arysize) = 2];
}


message Sheet_BattlepassTask
{
	repeated E_BattlepassTask E_BattlepassTask_List = 1[(yd_fieldoptions.field_arysize)=32];
}

message E_BattlepassShop
{
	optional int32 m_id = 1[(yd_fieldoptions.field_cname) = "id"];
	optional int32 m_group = 2[(yd_fieldoptions.field_cname) = "组id"];
	optional int32 m_itemID = 3[(yd_fieldoptions.field_cname) = "物品id"];
	optional int32 m_itemNum = 4[(yd_fieldoptions.field_cname) = "物品数量"];
	optional int32 m_costItem = 5[(yd_fieldoptions.field_cname) = "花费道具"];
	optional int32 m_costNum = 6[(yd_fieldoptions.field_cname) = "花费数量"];
	optional int32 m_numLimit = 7[(yd_fieldoptions.field_cname) = "限购数量"];
}


message Sheet_BattlepassShop
{
	repeated E_BattlepassShop E_BattlepassShop_List = 1[(yd_fieldoptions.field_arysize)=32];
}
